### PR TITLE
Max idle transform fix

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by jaheba** [comment]
Looking at the code, there is quite a lot of duplication.

**Review by lostella — APPROVED** [review]
LGTM, thanks! 🚀

**Comment by kashif** [comment]
do we need to update the pytorch estimators too?

**Comment by lostella** [comment]
> do we need to update the pytorch estimators too?

Probably yes 👍 not the same fix as here, but definitely setting the max idle transform in the base estimator class

File: src/gluonts/mx/model/estimator.py [hunk 1]
```diff
@@ -170,27 +172,34 @@ def train_model(
...
         cache_data: bool = False,
     ) -> TrainOutput:
+
         transformation = self.create_transformation()
 
-        transformed_training_data = transformation.apply(training_data)
+        with env._let(max_idle_transforms=max(len(training_data), 100)):
+            transformed_training_data = transformation.apply(training_data)
+            if cache_data:
+                transformed_training_data = Cached(transformed_training_data)
```
Comment (lostella) [inline src/gluonts/mx/model/estimator.py:181]: Here we are setting the `max_idle_transforms` when applying the transformation to the data.

File: src/gluonts/transform/_base.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ class FlatMapTransformation(Transformation):
 
     @validated()
     def __init__(self):
-        self.max_idle_transforms = max(env.max_idle_transforms, 100)
+        self.max_idle_transforms = env.max_idle_transforms
```
Comment (lostella) [inline src/gluonts/transform/_base.py:172]: However, here we are reading the value at construction time.

**Review by lostella — COMMENTED** [review]
~~@jaheba are we sure this is correct?~~ nevermind, it's correct: the instance splitter transformation gets constructed with the data loader

**Pull request merged**